bundles/framework/publisher2/view/PublisherSideBarHandler.js

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import { showModal } from 'oskari-ui/components/window';
+import { ValidationErrorMessage } from './dialog/ValidationErrorMessage';
+import { ReplaceConfirmDialogContent } from './dialog/ReplaceConfirmDialogContent';
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+import { GeneralInfoForm } from './form/GeneralInfoForm';
+import { PanelGeneralInfoHandler } from '../handler/PanelGeneralInfoHandler';
+import { MAP_SIZES, PanelMapPreviewHandler } from '../handler/PanelMapPreviewHandler';
+import { MapPreviewForm, MapPreviewTooltip } from './form/MapPreviewForm';
+import { mergeValues } from '../util/util';
+
+export const PUBLISHER_BUNDLE_ID = 'Publisher2';
+const PANEL_GENERAL_INFO_ID = 'panelGeneralInfo';
+const PANEL_MAPPREVIEW_ID = 'panelMapPreview';
+
+class PublisherSidebarUIHandler extends StateHandler {
+    constructor () {
+        super();
+        this.validationErrorMessageDialog = null;
+        this.replaceConfirmDialog = null;
+
+        this.generalInfoPanelHandler = new PanelGeneralInfoHandler();
+        this.mapPreviewPanelHandler = new PanelMapPreviewHandler();
+
+        this.state = {
+            collapseItems: []
+        };
+    }
+
+    updateGeneralInfoPanel () {
+        const newCollapseItems = this.getState().collapseItems.map(item => item);
+        const generalInfoPanel = newCollapseItems.find(item => item.key === PANEL_GENERAL_INFO_ID);
+        generalInfoPanel.children = this.renderGeneralInfoPanel();
+        this.updateState({
+            collapseItems: newCollapseItems
+        });
+    }
+
+    renderGeneralInfoPanel () {
```

Comment:
We should move the render functions to PublisherSidebar.js directly since they are always there. Maybe we need to do something like this with the tool panels though?

Reply:
Lets return to this once it's more ready. Probably easier to test things when jQuery panels are no longer in the way.